ui/panels-plugin/src/plugins/stat-chart/StatChartPanel.tsx

```diff
@@ -11,45 +11,101 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { JsonObject, AnyGraphQueryDefinition, PanelProps, useGraphQuery } from '@perses-ui/core';
+import {
+  AnyGraphQueryDefinition,
+  GraphData,
+  useGraphQuery,
+  JsonObject,
+  PanelProps,
+  usePanelState,
+} from '@perses-ui/core';
+import { Box, Skeleton } from '@mui/material';
 import { useMemo } from 'react';
 import { CalculationsMap, CalculationType } from '../../model/calculations';
-import { formatValue, UnitOptions } from '../../model/units';
+import { UnitOptions } from '../../model/units';
+import { ThresholdOptions, defaultThresholdInput } from '../../model/thresholds';
+import { StatChartData, StatChart } from '../../components/stat-chart/StatChart';
 
 export const StatChartKind = 'StatChart' as const;
 
 export type StatChartPanelProps = PanelProps<StatChartOptions>;
 
 interface StatChartOptions extends JsonObject {
+  name: string;
+  show_panel_header: string;
   query: AnyGraphQueryDefinition;
   calculation: CalculationType;
-  unit?: UnitOptions;
+  unit: UnitOptions;
+  thresholds?: ThresholdOptions;
+  sparkline?: {
+    show: boolean;
+  };
```

Comment:
Curious, what other options do you envision for `sparkline`? Would be nice if we didn't also need this boolean `show` property (i.e. if `sparkline` has an options object, then we infer that it should be shown rather than needing a separate boolean flag).

Reply:
Yeah true, I'll change this so an empty sparkline object is enough to show the line series and then [default colors](https://github.com/perses/perses/blob/61be6d5a3e3747818a98ba1d4f09b6b4114af979/ui/panels-plugin/src/components/stat-chart/StatChart.tsx#L79) can be overriden using something like: 
```
sparkline: {
  line_color?: string;
  line_width?: number;
  area_color?: string; // defaults to same as line_color
  // add more as needed
};
```
Which in StatChart would translate to [lineStyle.color](https://echarts.apache.org/en/option.html#series-line.lineStyle.color), lineStyle.width, areaStyle.color, etc. Also, any thoughts @jhardy?

Reply:
I [went ahead with the options above](https://github.com/perses/perses/pull/307/commits/a275bbd95597713be0b89c5ca1d770382e846f10) and also added line_opacity and area_opacity. I have some ideas about panel-level vs. chart-level props that I want to discuss with the team before going any further.